src/base/bittorrent/filterparserthread.cpp

```diff
@@ -37,75 +42,133 @@
 
 #include "base/logger.h"
 
+using namespace std::literals;
+
 namespace
 {
-    class IPv4Parser
+    struct IPFilterEntry
     {
-    public:
-        bool tryParse(const char *str)
-        {
-            unsigned char octetIndex = 0;
+        std::string_view range;
+        int accessLevel = 0;
+        // We don't care about this
+        // std::string_view description;
+    };
+
+    template <bool isDat> IPFilterEntry parseLine(std::string_view line)
```

Comment:
I think the template parameter is overused. Looking at the code it just splits into two distinctive code path, you really are writing two functions instead of one.